keel-core/src/test/kotlin/com/netflix/spinnaker/keel/services/ResourceStatusServiceTests.kt

```diff
@@ -214,12 +215,27 @@ class ResourceStatusServiceTests : JUnit5Minutests {
           }
 
           context("when last event is ResourceCheckUnresolvable") {
-            before {
-              events.add(ResourceCheckUnresolvable(resource, object : ResourceCurrentlyUnresolvable("") {}))
+            context("only created and unresolvable events") {
+              before {
+                events.clear()
+                events.add(ResourceCreated(resource))
+                events.add(ResourceCheckUnresolvable(resource, object : ResourceCurrentlyUnresolvable("") {}))
+              }
+
+              test("returns WAITING") {
+                expectThat(subject.getStatus(resource.id)).isEqualTo(WAITING)
+              }
             }
 
-            test("returns CURRENTLY_UNRESOLVABLE") {
-              expectThat(subject.getStatus(resource.id)).isEqualTo(CURRENTLY_UNRESOLVABLE)
+            context("more than created and unresolvable events") {
```

Comment:
You're missing a test for your `size < 5` condition if you want to cover all cases.